### ISSUE TITLE
FeatureAlphaDropout doesn't drop channels for (C, D, H, W)

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#69960]
### 🐛 Describe the bug

Related #69801

`FeatureAlphaDropout` also has similar problem.

```python
import torch
import torch.nn as nn
from torch.testing._internal.common_utils import freeze_rng_state

m = nn.FeatureAlphaDropout(p=0.5)
input = torch.randn(2, 2, 2, 2)

with freeze_rng_state():
    output_unbatched = m(input)

with freeze_rng_state():
    output_batched = m(input.unsqueeze(0)).squeeze(0)

torch.testing.assert_allclose(output_batched, output_unbatched)
```

### Versions

Latest master

**Comment by ngimel** [comment]
The docs for FeatureAlphaDropout are wrong (they specify 4d or 5d input, where in fact FeatureAlphaDropout accepts any input that's 2d+. As with other functions that don't specify expected spatial dimensionality of inputs (unlike Dropout2d, e.g.), FeatureAlphaDropout has no way of distinguishing between inputs that have or don't have batch dimension.

**Comment by jbschlosser** [comment]
> We didn't agree on the general way of handling this situation, did we?

Not fully yet but, in our last couple frontend meetings, we started to converge around option 3 from https://github.com/pytorch/pytorch/issues/60585#issuecomment-978316538. To maintain the current behavior, `num_spatial_dims=-2` as a default could work (i.e. `input.dim() - 2` to leave out the batch and channel dims from the spatial count). Setting `num_spatial_dims=2` for this module would allow for `(C, H, W)` or `(N, C, H, W)` inputs.

A more flexible approach for this module could be to manually specify `channel_dim=1`, which allows the channel dim to be anywhere instead of assuming it's directly prior to the spatial dims.
